Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1093,7 +1095,25 @@ void FitPeaks::fitSpectrumPeaks(
 
     double expected_peak_pos = expected_peak_centers[peak_index];
 
-    // Determine how to set the starting value parameters
+    // clone peak function for each peak (need to do this so can
+    // set center and calc any parameters from xml)
+    auto peakfunction =
+        std::dynamic_pointer_cast<API::IPeakFunction>(m_peakFunction->clone());
+    peakfunction->setCentre(expected_peak_pos);
+    peakfunction->setMatrixWorkspace(m_inputMatrixWS, wi, 0.0, 0.0);
```

Comment:
If this is the first peak on the first spectrum, and user specifies the starting value of  peak parameters, can this call override it by overwriting the user specified parameters (set in `peakfunction`)?

Reply:
This is a good catch - presently the user specified values will be used (if supplied) unless the same peak cross spectrum is found (in which case the fwhm will be estimated also). If it is the first spectrum (so there exists no same peak cross-spectrum) or the neighbouring peak in the same spectrum has been fitted the calculated values will be overwritten which is not exactly what we want. I will change the logic of this so that if there is no neighbouring peak found AND no cross-spectrum peak found it will use the user supplied values (i.e. user supplied values are only used if no peak in the spectrum has been fitted and the same peak in a previous spectrum has not been fitted). In which case if no user specified parameters are supplied it will use the A, B and S calculated from the Parameters.xml.